landoapi/transplants.py

```diff
@@ -301,6 +303,43 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
+    if short_name == "mozilla-central":
```

Comment:
based on the conversation in the original bug, it seems central is the only (relevant?) repo that undergoes a code freeze

Reply:
We still shouldn't check the value of `shortName` to determine whether this check is relevant. We should use `shortName` to get the `Repo` object for the current env using something like this: https://github.com/mozilla-conduit/lando-api/blob/04a1f37078c2206d92cb06fa206542dd2a2ea8d7/landoapi/api/uplift.py#L48

Then check a flag on the object similar to just below: https://github.com/mozilla-conduit/lando-api/blob/04a1f37078c2206d92cb06fa206542dd2a2ea8d7/landoapi/api/uplift.py#L58

Then it will look like:

```python
all_repos = get_repos_for_env(current_app.config.get("ENVIRONMENT")
short_name = repo.get("fields", {}).get("shortName")
local_repo = all_repos.get(short_name)

if not local_repo.soft_code_freeze:
    return

# rest of the code here
```

Reply:
> We still shouldn't check the value of `shortName` to determine whether this check is relevant. We should use `shortName` to get the `Repo` object for the current env.

+1, and would also help with setting a URL for our testing repos (see [this comment](https://github.com/mozilla-conduit/lando-api/pull/213/files#r958801182))